hedera-state-validator/src/main/java/com/hedera/statevalidation/validators/servicesstate/AccountValidator.java

```diff
@@ -34,24 +47,52 @@ public class AccountValidator {
     final long TOTAL_tHBAR_SUPPLY = 5_000_000_000_000_000_000L;
 
     @Test
-    void validate(DeserializedSignedState deserializedState, Report report) throws IOException {
-        final MerkleNodeState servicesState =
+    void validate(DeserializedSignedState deserializedState, Report report) throws IOException, InterruptedException {
+        final MerkleNodeState merkleNodeState =
                 deserializedState.reservedSignedState().get().getState();
 
-        ReadableKVState<AccountID, Account> accounts =
-                servicesState.getReadableStates(TokenServiceImpl.NAME).get(V0490TokenSchema.ACCOUNTS_KEY);
+        final VirtualMap virtualMap = (VirtualMap) merkleNodeState.getRoot();
+        assertNotNull(virtualMap);
+
+        final ReadableEntityIdStore entityCounters =
+                new ReadableEntityIdStoreImpl(merkleNodeState.getReadableStates(EntityIdService.NAME));
+        final ReadableKVState<AccountID, Account> accounts =
+                merkleNodeState.getReadableStates(TokenServiceImpl.NAME).get(V0490TokenSchema.ACCOUNTS_KEY);
 
         assertNotNull(accounts);
-        log.debug("Number of accounts: {}", accounts.size());
+        assertNotNull(entityCounters);
+
+        final long numAccounts = entityCounters.numAccounts();
+        log.debug("Number of accounts: {}", numAccounts);
 
+        AtomicLong accountsCreated = new AtomicLong(0L);
         AtomicLong totalBalance = new AtomicLong(0L);
-        accounts.keys().forEachRemaining(key -> {
-            final var value = accounts.get(key);
-            long tinybarBalance = value.tinybarBalance();
-            assertTrue(tinybarBalance >= 0);
-            totalBalance.addAndGet(tinybarBalance);
-        });
+
+        final int targetStateId = stateIdFor(TokenServiceImpl.NAME, V0490TokenSchema.ACCOUNTS_KEY);
+
+        InterruptableConsumer<Pair<Bytes, Bytes>> handler = pair -> {
+            final Bytes keyBytes = pair.left();
+            final int readStateId = readNextFieldNumber(keyBytes.toReadableSequentialData());
```

Comment:
A comment here would be nice to have to describe why this works

Reply:
I extracted this to a utility method